packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -850,6 +852,12 @@ export function diffProperties(
         // to update this element.
         updatePayload = [];
       }
+    } else if (
+      typeof nextProp === 'object' &&
+      nextProp !== null &&
+      nextProp.$$typeof === REACT_OPAQUE_OBJECT_TYPE
+    ) {
+      throw nextProp.toString();
```

Comment:
I don't think I understand why diff props throws here.

I guess we only hit this case if the values differed? Which is never expected?

Reply:
Btw even if we intend to throw, we must never throw strings. They won't get proper stack traces in reporting.

Reply:
This throws here so that in concurrent mode we can rerender. We don't need to throw this, however, because nextProp.toString() itself should throw, so I'll remove the `throw`